sympy/vector/operators.py

```diff
@@ -17,12 +18,15 @@ def _get_coord_sys_from_expr(expr, coord_sys=None):
             useinstead="do not use it",
             deprecated_since_version="1.1"
         ).warn()
-    if expr is Vector.zero:
-        return None
-    elif expr == 0:
+
+    try:
+        coord_sys = list(expr.atoms(CoordSysCartesian))
+        if len(coord_sys) == 1:
```

Comment:
In the future we will have to deal with mixed coordinate systems. But it's OK for now, let's keep it like this temporarily.